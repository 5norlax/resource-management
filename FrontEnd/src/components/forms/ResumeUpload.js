import React from 'react';
import { CustomInput, FormGroup, Row, FormText } from 'reactstrap';

const ResumeUpload = ({ resume, handleResume }) => {
    return (
        <div>
            <Row>
            <FormGroup className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12">
                <CustomInput type="file" id="ResumeUpload" name="ResumeUpload" accept="application/pdf" value={resume} onChange={handleResume} />
                <FormText color="muted">
                    Please upload your most up to date resume in PDF format.
                </FormText>
            </FormGroup>
            </Row>
            <hr />
        </div>
    )
}

export default ResumeUpload