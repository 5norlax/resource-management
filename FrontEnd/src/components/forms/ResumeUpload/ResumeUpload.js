import React from 'react'
import { CustomInput, FormGroup, Row, FormText } from 'reactstrap'
import ResumeModal from '../../resourceTable/ResumeModal/ResumeModal'

export default React.memo(function ResumeUpload({ resume, handleResume, existingResume, id, FirstName, LastName }) {
  return (
    <div>
      <Row>
        <FormGroup className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12">
          <CustomInput type="file" id="ResumeUpload" name="ResumeUpload" accept="application/pdf" value={resume} onChange={handleResume} />
          <FormText color="muted">
                    Please upload your most up to date resume in PDF format.
          </FormText>
        </FormGroup>
        <FormGroup className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12">
          {existingResume && 
                    <div className="text-center">
                      <ResumeModal id={id} FirstName={FirstName} LastName={LastName} />
                      <FormText color="muted">
                            Last Uploaded Resume
                      </FormText>
                    </div>
          }
        </FormGroup>
      </Row>
      <hr />
    </div>
  )
})