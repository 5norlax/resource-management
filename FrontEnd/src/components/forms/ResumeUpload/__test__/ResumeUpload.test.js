import React from 'react';
import ReactDOM from 'react-dom';
import { render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import ResumeUpload from '../ResumeUpload';

afterEach(cleanup);

describe('Test ResumeUpload', () => {

  it('matches snapshot', () => {
    const tree = renderer.create(<ResumeUpload />).toJSON()
    expect(tree).toMatchSnapshot();
  });

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ResumeUpload />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

});