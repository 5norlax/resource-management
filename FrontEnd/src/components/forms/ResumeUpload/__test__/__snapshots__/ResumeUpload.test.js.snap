// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    className="row"
  >
    <div
      className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12 form-group"
    >
      <div
        className="custom-file"
      >
        <input
          accept="application/pdf"
          className="custom-file-input"
          id="ResumeUpload"
          name="ResumeUpload"
          onChange={[Function]}
          type="file"
        />
        <label
          className="custom-file-label"
          htmlFor="ResumeUpload"
        >
          Choose file
        </label>
      </div>
      <small
        className="form-text text-muted"
      >
        Please upload your most up to date resume in PDF format.
      </small>
    </div>
    <div
      className="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-xs-12 form-group"
    />
  </div>
  <hr />
</div>
`;
